### ISSUE TITLE
Prepare for and hold a pyright/Ruff correction day (suggested date 17th October 2024)

### ISSUE THREAD

**Issue by KathrynBaker** [ISISComputingGroup/IBEX#8527]
As a developer working on IBEX I want to clear out and correct the legacy pyright/ruff errors that are in the system now that we have instigated these checks as standard so that small/urgent changes don't lead to lots of fixing.

# Acceptance Criteria
- [x] Everyone who can spends as much time as they can on this day fixing these issues
- [x] The named person on this ticket provides a breakdown of the work needed to facilitate the team undertaking the task (this has to be done before the day is held)

# Notes
- The majoirty of the group are available for the whole of that day, making it a suitable choice for this kind of focussed day

**Comment by Tom-Willemsen** [comment]
To maximise the benefit of this day, I think we should focus on core/central pieces of code rather than beamline or device-specific stuff. Also try to target repos with higher churn as fixing issues in those repos will have biggest benefit.

Initial repos to target (we likely won't get through _all_ of these):
- [ ] Block server
  * This can likely be split between multiple people
- [ ] Genie_python
  * Some bits already done, remaining bits can be split between multiple people
- [ ] System tests
  * This can likely be split between multiple people
- [ ] Upgrade scripts
- [ ] Install scripts
- [ ] Database server
- [ ] Reflectometry server
- [ ] IOC test framework central utilities
- [ ] Device emulator central utilities
- [ ] Configchecker
- [ ] Top level EPICS scripts e.g. `check_scripts` (repo checks), `build_ioc_startups`

**Comment by GRyall** [comment]
I suggest also checking that the device generator generates devices that are compliant (my device from the old generator is not, apologies if this has already been done)

**Comment by LowriJenkins** [comment]
Some of the genie python files will be done by https://github.com/ISISComputingGroup/genie_python/pull/439/files as well

**Comment by davidkeymer** [comment]
Some changes in the Config Checker repository: [PR](https://github.com/ISISComputingGroup/InstrumentChecker/pull/58)

**Comment by LowriJenkins** [comment]
[DatabaseServer](https://github.com/ISISComputingGroup/EPICS-inst_servers/pull/399)
[Required Genie Python File](https://github.com/ISISComputingGroup/genie_python/pull/442)

**Comment by LowriJenkins** [comment]
Do we want to propose a follow up to this ticket to deal with the area's not dealt with during this one? i.e. Blockserver, system tests, reflectometry server, and top level epics scripts?

**Comment by GRyall** [comment]
Follow on ticket: https://github.com/ISISComputingGroup/IBEX/issues/8553

**Comment by GRyall** [comment]
Remaining PRs for review before this issue can be closed:

- [ ] https://github.com/ISISComputingGroup/IBEX-device-generator/pull/4
- [ ] https://github.com/ISISComputingGroup/EPICS-upgrade/pull/83

All others have been reviewed and merged.

**Comment by Chsudeepta** [comment]
There are a number of repos which haven't been touched. Also for upgrade, when I created the PR, I wasn't aware that the ruleset I was using and the one builds are using aren't same. So I am doing more change to Upgrade (So the PR is really not ready).

**Comment by GRyall** [comment]
>There are a number of repos which haven't been touched.

Yes - that is why there is a follow on ticket. I think this ticket can be considered complete when the work done for the day has been reviewed and merged. 

>So the PR is really not ready

Understood - my comment was a holding one so it is clear at a glance why the ticket is still in review, not a chase.

**Comment by Chsudeepta** [comment]
The PR for the upgrade module is now available for review --> [PR_UPGRADE](https://github.com/ISISComputingGroup/EPICS-upgrade/pull/83)

**Comment by GRyall** [comment]
The device generator one has outstanding comments. I briefly looked at the epics-upgrade one, but am not actively reviewing it.

**Comment by LowriJenkins** [comment]
I'll have a look at the epics upgrade one then

**Comment by LowriJenkins** [comment]
Just device generator left to do

**Comment by GRyall** [comment]
completion of https://github.com/ISISComputingGroup/IBEX-device-generator/pull/4 has been moved into #8553. All other PRs have been reviewed, so closing this ticket.
